internal/httphandler/data_request_handler.go

```diff
@@ -144,3 +145,40 @@ func (d *dataRequestHandler) dataTransaction(response http.ResponseWriter, reque
 
 	d.txHandler.handleTransaction(response, txEnv, timeout)
 }
+
+func (d *dataRequestHandler) dataJSONQuery(response http.ResponseWriter, request *http.Request) {
+	payload, respondedErr := extractVerifiedQueryPayload(response, request, constants.PostDataQuery, d.sigVerifier)
+	if respondedErr {
+		return
```

Comment:
Shouldn't you reply here back to the client with the HTTP error?

Reply:
Hi Artem, How are you doing?

`extractVerifiedQueryPayload()` has sent the error already and that's why it is returning `respondedErr`. This design has been in play since the beginning.

Reply:
Doing good ;) 

> extractVerifiedQueryPayload() has sent the error already, and that's why it is returning respondedErr. This design has been in play since the beginning.

Too long haven't looked at the code and forgot it :)

PS. How are you doing?